backend/apps/challenges/views.py

```
@@ -354,11 +355,27 @@ def post(self, request, *args, **kwargs):
         cotd = get_object_or_404(ChallengeOfTheDay, date=today)
 
+        # Record this challenge completion as a learning activity so the
+        # active streak (and its XP multiplier) reflects today's work.
+        try:
+            from apps.progress.streak_engine import StreakEngine
+
+            streak_data = StreakEngine.record_activity(
+                request.user, activity_date=today
+            )
+            current_streak = streak_data.get("current_streak", 0)
+        except Exception:
```

Ruff (0.16.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)